samples/nrf9160/http_application_update/src/main.c

```diff
@@ -170,7 +274,31 @@ static int application_init(void)
 
 void main(void)
 {
-	int err;
+	int err = bsd_init();
+	switch (err) {
+	case MODEM_DFU_RESULT_OK:
+		printk("Modem firmware update successful!\n");
+		printk("Modem will run the new firmware after reboot\n");
+		k_thread_suspend(k_current_get());
+		break;
+	case MODEM_DFU_RESULT_UUID_ERROR:
+		printk("Modem firmware uuid error!\n");
+		break;
+	case MODEM_DFU_RESULT_AUTH_ERROR:
+		printk("Modem firmware update failed\n");
+		printk("Modem will run non-updated firmware on reboot.\n");
+		break;
+	case MODEM_DFU_RESULT_HARDWARE_ERROR:
+		printk("Modem firmware hardware error!\n");
+		break;
+	case MODEM_DFU_RESULT_INTERNAL_ERROR:
```

Comment:
That's a fatal error, the modem is bricked.

Reply:
Keeping it as is? No way of handling this in a library? Is it okei for the sample to have this or should it be extracted out as it would be common for samples which supports modem firmware upgrade